src/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.hpp

```diff
@@ -0,0 +1,259 @@
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/MakeArray.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace AnalyticData {
+
+/*!
+ * \brief Initial data for axially symmetric Bondi-Hoyle accretion.
+ *
+ * In the context of studying Bondi-Hoyle accretion, i.e. non-spherical
+ * accretion on to a moving Kerr black hole, this class implements the method
+ * proposed by Font & Ibáñez (1998) to initialize the GRMHD variables.
+ * The fluid quantities are initialized with their (constant) values far
+ * from the black hole, e.g. \f$\rho = \rho_\infty\f$. Here we assume a
+ * polytropic equation of state, so only the rest mass density, as well as the
+ * polytropic constant and the polytropic exponent, are provided as inputs.
+ * The spatial velocity is initialized using a field that ensures that the
+ * injected gas reproduces a continuous parallel wind at large distances.
+ * The direction of this flow is chosen to be along the black hole spin.
+ * In spherical Kerr-Schild coordinates,
+ *
+ * \f{align*}
+ * v^r &= \frac{1}{\sqrt{\gamma_{rr}}}v_\infty \cos\theta\\
+ * v^\theta &= -\frac{1}{\sqrt{\gamma_{\theta\theta}}}v_\infty \sin\theta\\
+ * v^\phi &= 0.
+ * \f}
+ *
+ * where \f$\gamma_{ij} = g_{ij}\f$ is the spatial metric, and \f$v_\infty\f$
+ * is the flow speed far from the black hole.
+ * Finally, the magnetic field is initialized using Wald's solution to Maxwell's
+ * equations in Kerr black hole spacetime. In spherical Kerr-Schild coordinates,
+ * the spatial components of the Faraday tensor read
+ *
+ * \f{align*}
+ * F_{r\theta} &= a B_0 \left[1 + \frac{2Mr}{\Sigma^2}(r^2 - a^2)\right]
+ * \sin\theta\cos\theta\\
+ * F_{\theta\phi} &= B_0\left[\Delta +
+ * \frac{2Mr}{\Sigma^2}(r^4 - a^4)\right]\sin\theta\cos\theta\\
+ * F_{\phi r} &= - B_0\left[r + \frac{M a^2}{\Sigma^2}
+ * (r^2 - a^2\cos^2\theta)(1 + \cos^2\theta)\right]\sin^2\theta.
+ * \f}
+ *
+ * where \f$\Sigma = r^2 + a^2\cos^2\theta\f$ and \f$\Delta = r^2 - 2Mr +
+ * a^2\f$. The associated Eulerian magnetic field is
+ *
+ * \f{align*}
+ * B^r = \frac{F_{\theta\phi}}{\sqrt\gamma},\quad
+ * B^\theta = \frac{F_{\phi r}}{\sqrt\gamma},\quad
+ * B^\phi = \frac{F_{r\theta}}{\sqrt\gamma}.
+ * \f}
+ *
+ * where \f$\gamma = \text{det}(\gamma_{ij})\f$. Wald's solution reproduces a
+ * uniform magnetic field far from the black hole.
+ */
+class BondiHoyleAccretion {
+ public:
+  /// The mass of the black hole.
+  struct BlackHoleMass {
+    using type = double;
+    static constexpr OptionString help = {"The mass of the black hole."};
+  };
+  /// The black hole spin magnitude in units of the black hole mass.
+  struct BlackHoleSpin {
+    using type = double;
+    static constexpr OptionString help = {
+        "The dimensionless black hole spin magnitude."};
```

Comment:
Still confused what spin is requested here. Please use a name instead of BlackHoleSpin to clarify, such as DimensionlessBlackHoleSpin, if indeed you want the dimensionless spin (though the comment on line 84 seems to suggest you want DimensionlessBlackHoleSpinTimesMass instead?) If possible, I prefer sticking with mass and dimensionless spin.